Application/src/visuals/ActorClass.java

```diff
@@ -0,0 +1,93 @@
+package visuals;
+
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.paint.Color;
+
+
+/**
+ * An ActorClass wrapper extending DiagramClass. Differs
+ * from the latter by images used and canvas position.
+ * @author Kosara Golemshinska
+ * @version 1.1
+ */
+public class ActorClass extends DiagramClass {
+
+    // The images retrieved for the actor class.
+    private static Image king1 = new Image("resources/king_cape_up.png");
+    private static Image king2 = new Image("resources/king_cape_down.png");
+
+    // The platform for the actor class.
+    private static Image platform = new Image("resources/carpet_platform.png");
+
+    // The 2 states of the king/actor.
+    private Image[] kingStates = {king1, king2};
+    // Index used to write the animation state.
+    private double kingAnimIndex = 0.0;
+
+    /**
+     * Default constructor, calls the super constructor.
+     * @param name the class name.
+     */
+    ActorClass(String name) {
+        super(name);
+    }
+
+    /**
+     * From: Renderable
+     * Increments the index to use when accessing
+     * the king states array.
+     */
+    @Override
+    public void update() {
+        kingAnimIndex += 0.25;
+        if (kingAnimIndex >= kingStates.length)
+            kingAnimIndex = 0;
+    }
+
+    /**
+     * Renders the ActorClass on the canvas.
+     * @param gc GraphicalContext for a canvas.
+     */
+    @Override
+    public void render(GraphicsContext gc) {
```

Comment:
This is still just duplicate code but if everyone else is fine with it then I guess we can have it as technical debt

Reply:
It isn't though, it differs, so I had to override it. If you look at line 82, I had to change the last 2 arguments to fit the king image onto the platform.

Reply:
its 99% the same

Reply:
so what's your suggestion?

Reply:
this: "This is still just duplicate code but if everyone else is fine with it then I guess we can have it as technical debt"

Reply:
okay, we can refactor next sprint, I suppose